#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/robotstadium/protos/RobotstadiumSoccerField.proto"
EXTERNPROTO "../protos/RobocupSoccerBall.proto"
EXTERNPROTO "../protos/NaoV6.proto"
EXTERNPROTO "../protos/NaoJersey.proto"

WorldInfo {
  optimalThreadCount 8
  gpsCoordinateSystem "WGS84"
}
Viewpoint {
  orientation -0.26464051364050023 0.38187041188345017 0.8855170168145029 1.0305212734913225
  position -5.391490937708016 -6.958171685875335 4.387398536464877
}
DEF BALL RobocupSoccerBall {
  hidden linearVelocity_0 0.04850750594987408 -0.0036706586151751407 6.413788345650602e-05
  hidden angularVelocity_0 0.05246163314068737 0.6933808753190291 0.006047318691602785
  translation -3.6659416074253617 -0.2275521937262738 0.06967808113908427
  rotation 0.9633755928629015 -0.20470379869949873 0.17321611320557995 1.756852594601802
  name "RobocupSoccerBall"
}
DEF ATTACKER_1 NaoV6 {
  hidden translation_11 0 0 0.1265
  hidden rotation_11 2.039391118111131e-15 -0.9999999999999923 1.2365842878574355e-07 2.9802322387695313e-08
  hidden translation_18 0 -0.098 0.1
  hidden rotation_18 1.7746810938652007e-12 0.9999999999736582 7.258348486194927e-06 5.260088794926779e-06
  hidden translation_19 0.105 -0.015 0
  hidden rotation_19 1 0 0 0
  hidden translation_20 0 0 0
  hidden rotation_20 0 0 1 0
  hidden translation_21 0.05595 0 0
  hidden rotation_21 -1 0 0 2.1696418193818693e-07
  hidden translation_22 0.069 -0.013 -0.005
  hidden translation_23 0.014 0 0
  hidden rotation_23 0 1 0 0.9599999993871381
  hidden translation_24 0.014 0 0
  hidden rotation_24 0 1 0 0.9599999997189943
  hidden translation_25 0.069 0.013 -0.005
  hidden translation_26 0.014 0 0
  hidden rotation_26 0 0.9999999999999999 0 0.9599999993871304
  hidden translation_27 0.014 0 0
  hidden rotation_27 0 1 0 0.9599999997190058
  hidden translation_28 0.049 0 -0.026
  hidden rotation_28 0 1 0 0
  hidden translation_29 0.014 0 0
  hidden rotation_29 0 -1 0 0.9599999998331861
  hidden translation_30 0 0.098 0.1
  hidden rotation_30 -1.778822936627131e-12 0.9999999999736705 -7.2566471212226395e-06 5.260088794926779e-06
  hidden translation_31 0.105 0.015 0
  hidden rotation_31 1 0 0 0
  hidden translation_32 0 0 0
  hidden rotation_32 0 0 1 0
  hidden translation_33 0.05595 0 0
  hidden rotation_33 1 0 0 2.1696418193818693e-07
  hidden translation_34 0.069 -0.013 -0.005
  hidden translation_35 0.014 0 0
  hidden rotation_35 0 0.9999999999999999 0 0.9599999993870804
  hidden translation_36 0.014 0 0
  hidden rotation_36 0 1 0 0.9599999997190203
  hidden translation_37 0.069 0.013 -0.005
  hidden translation_38 0.014 0 0
  hidden rotation_38 0 1 0 0.9599999993870905
  hidden translation_39 0.014 0 0
  hidden rotation_39 0 0.9999999999999999 0 0.9599999997190006
  hidden translation_40 0.049 0 -0.026
  hidden rotation_40 0 1 0 0
  hidden translation_41 0.014 0 0
  hidden rotation_41 0 -1 0 0.9599999998331572
  hidden translation_42 0 -0.05 -0.085
  hidden rotation_42 0 -0.7071067811865476 -0.7071067811865476 2.1001652392683837e-06
  hidden translation_43 0 0 0
  hidden rotation_43 -1 0 0 4.2467002262954494e-06
  hidden translation_44 0 0 0
  hidden rotation_44 0 -1 0 2.4775393152376715e-06
  hidden translation_45 0 0 -0.1
  hidden rotation_45 0 -1 0 2.5071176031886915e-06
  hidden translation_46 0 0 -0.1029
  hidden rotation_46 0 -1 0 2.5776855428421745e-06
  hidden translation_47 0 0 0
  hidden rotation_47 -1 0 0 1.933020075342912e-06
  hidden translation_52 0 0.05 -0.085
  hidden rotation_52 0 -0.7071067811865475 0.7071067811865475 2.1001652392683837e-06
  hidden translation_53 0 0 0
  hidden rotation_53 1 0 0 4.2467002262954494e-06
  hidden translation_54 0 0 0
  hidden rotation_54 0 -1 0 2.4775393152376715e-06
  hidden translation_55 0 0 -0.1
  hidden rotation_55 0 -1 0 2.5071176031886915e-06
  hidden translation_56 0 0 -0.1029
  hidden rotation_56 0 -1 0 2.5776855428421745e-06
  hidden translation_57 0 0 0
  hidden rotation_57 1 0 0 1.933020075342912e-06
  translation -0.8251483377159818 4.1876888414713775e-05 0.33251997074027595
  rotation -0.00012720604473901936 0.1915373144411161 0.9814853238817721 0.001365636540488064
  name "ATTACKER_1"
  controller "SupervisorPositionController"
  controllerArgs [
    "1"
    "0"
  ]
  supervisor TRUE
  synchronization FALSE
  jersey NaoJersey {
    jerseyTexture [
      "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/softbank/nao/protos/textures/nao_2_red.png"
    ]
  }
}
DEF DEFENDER_1 NaoV6 {
  hidden translation_11 0 0 0.1265
  hidden rotation_11 8.94593964857922e-10 -0.9989762480916807 0.04523776905053074 4.2146848510894035e-08
  hidden translation_18 0 -0.098 0.1
  hidden rotation_18 -2.6021709249531785e-09 0.9995666627548577 -0.029436146302063638 5.235887231642763e-06
  hidden translation_19 0.105 -0.015 0
  hidden rotation_19 -1 0 0 5.575503985246947e-07
  hidden translation_20 0 0 0
  hidden rotation_20 0 0 -1 5.65459309846471e-07
  hidden translation_21 0.05595 0 0
  hidden rotation_21 1 0 0 0
  hidden translation_22 0.069 -0.013 -0.005
  hidden rotation_22 0 1 0 0.9600000001008537
  hidden translation_23 0.014 0 0
  hidden rotation_23 0 1 0 0.9599999993835447
  hidden translation_24 0.014 0 0
  hidden rotation_24 0 0.9999999999999999 0 0.959999999721449
  hidden translation_25 0.069 0.013 -0.005
  hidden rotation_25 0 0.9999999999999999 0 0.9599999999634496
  hidden translation_26 0.014 0 0
  hidden rotation_26 0 1 0 0.959999999382585
  hidden translation_27 0.014 0 0
  hidden rotation_27 0 1 0 0.9599999997210139
  hidden translation_28 0.049 0 -0.026
  hidden rotation_28 0 1 0 0
  hidden translation_29 0.014 0 0
  hidden rotation_29 0 -1 0 0.9599999998323787
  hidden translation_30 0 0.098 0.1
  hidden rotation_30 -2.969810075174905e-09 0.9995533271570473 -0.029885551179075372 5.258315546227758e-06
  hidden translation_31 0.105 0.015 0
  hidden rotation_31 -1 0 0 5.101335873789709e-07
  hidden translation_32 0 0 0
  hidden rotation_32 0 0 -0.9999999999999999 6.286806794920288e-07
  hidden translation_33 0.05595 0 0
  hidden rotation_33 -0.9999999999999999 0 0 1.976862424823881e-07
  hidden translation_34 0.069 -0.013 -0.005
  hidden rotation_34 0 1 0 0.9600000001022958
  hidden translation_35 0.014 0 0
  hidden rotation_35 0 1 0 0.9599999993753511
  hidden translation_36 0.014 0 0
  hidden rotation_36 0 0.9999999999999999 0 0.959999999717747
  hidden translation_37 0.069 0.013 -0.005
  hidden rotation_37 0 1 0 0.9599999999630082
  hidden translation_38 0.014 0 0
  hidden rotation_38 0 1 0 0.959999999374401
  hidden translation_39 0.014 0 0
  hidden rotation_39 0 0.9999999999999999 0 0.9599999997172873
  hidden translation_40 0.049 0 -0.026
  hidden rotation_40 0 1 0 0
  hidden translation_41 0.014 0 0
  hidden rotation_41 0 -0.9999999999999999 0 0.9599999998301822
  hidden translation_42 0 -0.05 -0.085
  hidden rotation_42 0 -0.7071067811865475 -0.7071067811865475 1.0053591605087028e-06
  hidden translation_43 0 0 0
  hidden rotation_43 -1 0 0 0.19198402940384757
  hidden translation_44 0 0 0
  hidden rotation_44 0 -1 0 0.5953756352324437
  hidden translation_45 0 0 -0.1
  hidden rotation_45 0 1 0 1.3081169323062862
  hidden translation_46 0 0 -0.1029
  hidden rotation_46 0 -1 0 0.7127441911698682
  hidden translation_47 0 0 0
  hidden rotation_47 1 0 0 0.19197797328385255
  hidden translation_52 0 0.05 -0.085
  hidden rotation_52 0 -0.7071067811865476 0.7071067811865476 9.437318433433038e-06
  hidden translation_53 0 0 0
  hidden rotation_53 -1 0 0 0.12207343976429323
  hidden translation_54 0 0 0
  hidden rotation_54 0 -0.9999999999999999 0 0.4311370155687816
  hidden translation_55 0 0 -0.1
  hidden rotation_55 0 1 0 1.0248686827291622
  hidden translation_56 0 0 -0.1029
  hidden rotation_56 0 -1 0 0.5937504028865658
  hidden translation_57 0 0 0
  hidden rotation_57 1 0 0 0.17272374985620348
  translation -1.866214782634337 0.9756835744016563 0.3023512691259693
  rotation 0.9868527282954227 0.07423794902587007 0.14356329468326043 0.13427577523000142
  name "DEFENDER_1"
  controller "<none>"
  controllerArgs [
    "1"
    "0"
  ]
  supervisor TRUE
  synchronization FALSE
  jersey NaoJersey {
    jerseyTexture [
      "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/softbank/nao/protos/textures/nao_2_red.png"
    ]
  }
}
DEF DEFENDER_2 NaoV6 {
  hidden translation_11 0 0 0.1265
  hidden rotation_11 -0.00024358571843509263 -0.9993515636670877 -0.03600545600522922 0.01353909806784224
  hidden translation_18 0 -0.098 0.1
  hidden rotation_18 -0.004416845142985715 0.567327875291554 -0.8234801596858017 0.018908025928656837
  hidden translation_19 0.105 -0.015 0
  hidden rotation_19 -1 0 0 3.4240261593687747e-07
  hidden translation_20 0 0 0
  hidden rotation_20 0 0 0.9999999999999999 6.467859272423625e-07
  hidden translation_21 0.05595 0 0
  hidden rotation_21 -1 0 0 1.4600096599955432e-07
  hidden translation_22 0.069 -0.013 -0.005
  hidden translation_23 0.014 0 0
  hidden rotation_23 0 1 0 0.9599999993550103
  hidden translation_24 0.014 0 0
  hidden rotation_24 0 1 0 0.9599999997167223
  hidden translation_25 0.069 0.013 -0.005
  hidden translation_26 0.014 0 0
  hidden rotation_26 0 1 0 0.9599999993549738
  hidden translation_27 0.014 0 0
  hidden rotation_27 0 1 0 0.9599999997167199
  hidden translation_28 0.049 0 -0.026
  hidden rotation_28 0 1 0 0
  hidden translation_29 0.014 0 0
  hidden rotation_29 0 -1 0 0.9599999998254156
  hidden translation_30 0 0.098 0.1
  hidden rotation_30 -0.0028199044581510495 0.34516303107669294 -0.9385385075300837 0.017409307790232583
  hidden translation_31 0.105 0.015 0
  hidden rotation_31 -1 0 0 3.4240261593687747e-07
  hidden translation_32 0 0 0
  hidden rotation_32 0 0 -1 8.429369702178807e-08
  hidden translation_33 0.05595 0 0
  hidden rotation_33 1 0 0 1.0745380149674385e-07
  hidden translation_34 0.069 -0.013 -0.005
  hidden rotation_34 0 1 0 0.9600000000938723
  hidden translation_35 0.014 0 0
  hidden rotation_35 0 1 0 0.9599999993595437
  hidden translation_36 0.014 0 0
  hidden rotation_36 0 1 0 0.9599999997170388
  hidden translation_37 0.069 0.013 -0.005
  hidden rotation_37 0 1 0 0.9600000000938805
  hidden translation_38 0.014 0 0
  hidden rotation_38 0 1 0 0.9599999993595226
  hidden translation_39 0.014 0 0
  hidden rotation_39 0 1 0 0.959999999717033
  hidden translation_40 0.049 0 -0.026
  hidden rotation_40 0 1 0 0
  hidden translation_41 0.014 0 0
  hidden rotation_41 0 -1 0 0.9599999998264307
  hidden translation_42 0 -0.05 -0.085
  hidden rotation_42 0 -0.7071067811865476 -0.7071067811865476 0.016345157767634566
  hidden translation_43 0 0 0
  hidden rotation_43 -0.9999999999999999 0 0 0.08378122089466725
  hidden translation_44 0 0 0
  hidden rotation_44 0 -1 0 8.381979332689972e-06
  hidden translation_45 0 0 -0.1
  hidden rotation_45 0 -1 0 5.552277439879193e-06
  hidden translation_46 0 0 -0.1029
  hidden rotation_46 0 -1 0 6.251174764120599e-06
  hidden translation_47 0 0 0
  hidden rotation_47 1 0 0 1.2758875086655164e-05
  hidden translation_52 0 0.05 -0.085
  hidden rotation_52 0 -0.7071067811865476 0.7071067811865476 0.21659029901227045
  hidden translation_53 0 0 0
  hidden rotation_53 1 0 0 0.3050865098038872
  hidden translation_54 0 0 0
  hidden rotation_54 0 1 0 2.4257332974822378e-06
  hidden translation_55 0 0 -0.1
  hidden rotation_55 0 1 0 2.7126662542997553e-06
  hidden translation_56 0 0 -0.1029
  hidden rotation_56 0 1 0 3.240651421325129e-06
  hidden translation_57 0 0 0
  hidden rotation_57 1 0 0 1.810107784159271e-06
  translation -0.08339595019294113 0.5116973758113217 0.33364753432774924
  rotation 0.6622586204374118 0.5724567237327891 0.48343233146890097 0.046901920076105434
  name "DEFENDER_2"
  controller "<none>"
  controllerArgs [
    "1"
    "0"
  ]
  supervisor TRUE
  synchronization FALSE
  jersey NaoJersey {
    jerseyTexture [
      "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/softbank/nao/protos/textures/nao_2_red.png"
    ]
  }
}
DEF GOALKEEPER_1 NaoV6 {
  hidden linearVelocity_0 0.03240473027114688 -0.008652044374175678 0.007365746556798128
  hidden angularVelocity_0 0.009330708195034803 -0.09412129079463862 -0.03570028379181498
  hidden translation_11 0 0 0.1265
  hidden rotation_11 -4.2489306623653805e-10 -0.9997150115532811 -0.023872487826425863 2.9802322387695313e-08
  hidden linearVelocity_11 0.02063593768544912 -0.00993069473129849 0.0076636430106026675
  hidden angularVelocity_11 0.009330716332893932 -0.09412132564151506 -0.035700283251739094
  hidden translation_18 0 -0.098 0.1
  hidden rotation_18 -0.17163142852313468 0.9754240012365206 -0.1381689927402105 1.8159444283726218
  hidden linearVelocity_18 0.002200162435712764 -0.007566942633719058 0.007738781983310302
  hidden angularVelocity_18 0.09109353781464799 1.30182225109176 -0.07921646332771805
  hidden translation_19 0.105 -0.015 0
  hidden rotation_19 1 0 0 1.3959999323301258
  hidden linearVelocity_19 -0.0650470704687446 -0.0018875347609014909 0.023293860298181833
  hidden angularVelocity_19 0.09109352390298953 1.301822302995372 -0.07921650478555994
  hidden translation_20 0 0 0
  hidden rotation_20 0 0 1 1.3959554958554201
  hidden linearVelocity_20 -0.11503017556794105 0.00023691325191293447 0.0005315496364765374
  hidden angularVelocity_20 0.09110294307757699 1.301202611290665 -0.07898870296327787
  hidden translation_21 0.05595 0 0
  hidden rotation_21 -0.9999999999999999 0 0 4.214684851089411e-07
  hidden linearVelocity_21 -0.144713772926427 -0.002508190473275005 -0.07855651921516182
  hidden angularVelocity_21 0.09110203879146991 1.3012025994980008 -0.07898832856797083
  hidden translation_22 0.069 -0.013 -0.005
  hidden rotation_22 0 1 0 0.9600000003418018
  hidden linearVelocity_22 -0.1855086873644597 -0.0017543292303323838 -0.11322389652217936
  hidden angularVelocity_22 0.09110203929692894 1.3012026000823125 -0.07898832721867652
  hidden translation_23 0.014 0 0
  hidden rotation_23 0 1 0 0.9599999993035301
  hidden linearVelocity_23 -0.19113480001926508 -0.0013015519341097494 -0.1122902304968934
  hidden angularVelocity_23 0.09110203875437688 1.3012025995095955 -0.07898832851138829
  hidden translation_24 0.014 0 0
  hidden rotation_24 0 0.9999999999999999 0 0.9599999997664385
  hidden linearVelocity_24 -0.18973447485500722 -0.0006035654459557683 -0.09925450189562032
  hidden angularVelocity_24 0.09110203882269559 1.3012025995791137 -0.07898832834118832
  hidden translation_25 0.069 0.013 -0.005
  hidden rotation_25 0 1 0 0.9600000003413224
  hidden linearVelocity_25 -0.15587238878934329 -0.004561015733480963 -0.124962955704865
  hidden angularVelocity_25 0.09110203928743368 1.3012026000728132 -0.07898832724041853
  hidden translation_26 0.014 0 0
  hidden rotation_26 0 0.9999999999999999 0 0.9599999993076744
  hidden linearVelocity_26 -0.1614985007187236 -0.004108238511340089 -0.12402928980651765
  hidden angularVelocity_26 0.09110203874437552 1.3012025994969434 -0.07898832853982
  hidden translation_27 0.014 0 0
  hidden rotation_27 0 1 0 0.9599999997660037
  hidden linearVelocity_27 -0.16009817519158107 -0.003410252059939502 -0.11099356126878941
  hidden angularVelocity_27 0.09110203881209801 1.3012025995662873 -0.07898832837048833
  hidden translation_28 0.049 0 -0.026
  hidden rotation_28 0 1 0 0
  hidden linearVelocity_28 -0.16697034509964553 -0.0018397430714228902 -0.09324070462796244
  hidden angularVelocity_28 0.09110203851604118 1.3012025992198308 -0.0789883292537521
  hidden translation_29 0.014 0 0
  hidden rotation_29 0 -1 0 0.9599999998722856
  hidden linearVelocity_29 -0.1713834372600626 -0.0024543286585272336 -0.10837973136990511
  hidden angularVelocity_29 0.09110203845645115 1.3012025991555876 -0.07898832941087988
  hidden translation_30 0 0.098 0.1
  hidden rotation_30 0.17220752677387166 0.9786650618745496 0.11206812387342821 2.012720909556342
  hidden linearVelocity_30 0.02373367181367276 -0.00860801231065076 0.011256708799265533
  hidden angularVelocity_30 0.04159894658244587 0.4569239811991818 -0.052813362312923315
  hidden translation_31 0.105 0.015 0
  hidden rotation_31 -1 0 0 1.3959999705225958
  hidden linearVelocity_31 0.004083372797915129 -0.005537883798048619 0.022271553084872383
  hidden angularVelocity_31 0.0415991667391714 0.45692439368252497 -0.05281337932906382
  hidden translation_32 0 0 0
  hidden rotation_32 0 0 -1 0.9813189531051304
  hidden linearVelocity_32 0.03048842824865872 -0.026393578582375544 0.04236811175002124
  hidden angularVelocity_32 -0.7141295391440808 -1.5266221119721912 -0.5112499365863468
  hidden translation_33 0.05595 0 0
  hidden rotation_33 1 0 0 2.8115454709698594e-07
  hidden linearVelocity_33 0.1139370748779509 -0.08394115117378717 0.09726315987992588
  hidden angularVelocity_33 -0.7141296612132825 -1.5266220016548508 -0.5112495741116723
  hidden translation_34 0.069 -0.013 -0.005
  hidden rotation_34 0 1 0 0.9599999999376402
  hidden linearVelocity_34 0.1519131889436815 -0.11824268961098194 0.14634862250659333
  hidden angularVelocity_34 -0.7141296611428104 -1.5266220015457583 -0.5112495744715297
  hidden translation_35 0.014 0 0
  hidden rotation_35 0 1 0 0.9599999993509544
  hidden linearVelocity_35 0.153545042214247 -0.12015279108791262 0.14975975302165226
  hidden angularVelocity_35 -0.7141296611556052 -1.526622001602382 -0.5112495746499276
  hidden translation_36 0.014 0 0
  hidden rotation_36 0 1 0 0.9599999997275154
  hidden linearVelocity_36 0.14083901633083332 -0.11236969063819781 0.1443145175891827
  hidden angularVelocity_36 -0.7141296611721737 -1.5266220016383019 -0.5112495746730737
  hidden translation_37 0.069 0.013 -0.005
  hidden rotation_37 0 1 0 0.9599999999356077
  hidden linearVelocity_37 0.17563177732019417 -0.11610857148245629 0.1070308930734264
  hidden angularVelocity_37 -0.7141296611395391 -1.5266220015490897 -0.5112495744649562
  hidden translation_38 0.014 0 0
  hidden rotation_38 0 0.9999999999999999 0 0.9599999993501577
  hidden linearVelocity_38 0.1772636310499072 -0.11801867280802279 0.11044202265302933
  hidden angularVelocity_38 -0.7141296611508433 -1.5266220016058263 -0.5112495746407772
  hidden translation_39 0.014 0 0
  hidden rotation_39 0 0.9999999999999999 0 0.959999999727439
  hidden linearVelocity_39 0.16455760539610637 -0.11023557228270714 0.1049967867527681
  hidden angularVelocity_39 -0.7141296611681157 -1.526622001640943 -0.5112495746640676
  hidden translation_40 0.049 0 -0.026
  hidden rotation_40 0 1 0 0
  hidden linearVelocity_40 0.13815503017282918 -0.10133366632242732 0.11526467764846643
  hidden angularVelocity_40 -0.7141296611427564 -1.5266220015119984 -0.5112495740288023
  hidden translation_41 0.014 0 0
  hidden rotation_41 0 -1 0 0.9599999998252743
  hidden linearVelocity_41 0.1541482046396557 -0.11157342491066181 0.12343836354756166
  hidden angularVelocity_41 -0.7141296611389958 -1.5266220014961864 -0.5112495739855522
  hidden translation_42 0 -0.05 -0.085
  hidden rotation_42 0 -0.7071067811865475 -0.7071067811865475 0.00015902544921899873
  hidden linearVelocity_42 0.04041247582972287 -0.007524257871220035 0.006454071256670734
  hidden angularVelocity_42 0.00942407518877641 -0.09239926191757687 -0.034082097780554425
  hidden translation_43 0 0 0
  hidden rotation_43 1 0 0 0.4053049622421032
  hidden linearVelocity_43 0.04296064435498826 -0.006977308230032267 0.005601683270372296
  hidden angularVelocity_43 0.004438687468790566 -0.09210411647169373 -0.03409773234817194
  hidden translation_44 0 0 0
  hidden rotation_44 0 -1 0 0.6202368629399118
  hidden linearVelocity_44 0.03039618293559256 -0.0029253227711178104 -0.0016246708295280445
  hidden angularVelocity_44 0.02502730361059342 0.26639701918435377 0.1070287090576778
  hidden translation_45 0 0 -0.1
  hidden rotation_45 0 1 0 1.0423287534726664
  hidden linearVelocity_45 0.00971121282947168 0.000597742080178436 -0.005240583266762842
  hidden angularVelocity_45 0.020844838183742217 0.19361192401453983 0.07837573103793233
  hidden translation_46 0 0 -0.1029
  hidden rotation_46 0 -1 0 0.4284434967623479
  hidden linearVelocity_46 -0.0002848013014638373 -0.0004913488029235021 -6.59167921623714e-05
  hidden angularVelocity_46 0.009420590090438606 -0.005288081959665174 7.51886245423604e-05
  hidden translation_47 0 0 0
  hidden rotation_47 -1 0 0 0.38799975570030626
  hidden linearVelocity_47 -7.736123655470087e-05 -0.00012715748616651617 4.4533053238152834e-05
  hidden angularVelocity_47 0.009420016466174911 -0.00528724353521713 7.53320417300372e-05
  hidden linearVelocity_48 -2.7940633834483827e-05 -4.698023013550084e-05 -1.6998802425557114e-05
  hidden angularVelocity_48 0.009419300025143079 -0.005286202787805446 7.541882156451503e-05
  hidden translation_52 0 0.05 -0.085
  hidden rotation_52 0 -0.7071067811865476 0.7071067811865476 0.00016552933705696899
  hidden linearVelocity_52 0.04340198352929833 -0.00764072485270619 0.007608726951154203
  hidden angularVelocity_52 0.009436286147962852 -0.09250660197288584 -0.03742181278760026
  hidden translation_53 0 0 0
  hidden rotation_53 1 0 0 0.4537038940633333
  hidden linearVelocity_53 0.046808126030496514 -0.007119835831899536 0.007089804789078587
  hidden angularVelocity_53 0.0029152877243423243 -0.09212759020886033 -0.03743916804239709
  hidden translation_54 0 0 0
  hidden rotation_54 0 -1 0 0.8868447147699396
  hidden linearVelocity_54 0.02340365816289807 0.007033358049759836 -0.02031153596993216
  hidden angularVelocity_54 0.04635198957379519 0.6689415031266088 0.30561773961172506
  hidden translation_55 0 0 -0.1
  hidden rotation_55 0 1 0 1.3057233458503656
  hidden linearVelocity_55 -0.06489012006717265 0.01450794907176883 -0.024003926997606378
  hidden angularVelocity_55 0.08092779620879255 1.2750402624705426 0.5788483556936396
  hidden translation_56 0 0 -0.1029
  hidden rotation_56 0 -1 0 0.4252266771566463
  hidden linearVelocity_56 -0.1327093114070808 0.0038287722825212857 0.008886093138467917
  hidden angularVelocity_56 0.007765522454795294 -0.0071273113838219975 0.0008704647461272841
  hidden translation_57 0 0 0
  hidden rotation_57 -0.9999999999999999 0 0 0.3977610099190647
  hidden linearVelocity_57 -0.1324405577945408 0.004149633899512908 0.009108920559864815
  hidden angularVelocity_57 0.00776552135862367 -0.007127283467430773 0.000870468038233574
  hidden linearVelocity_58 -0.13238442514221027 0.00421070564100288 0.009105520758917952
  hidden angularVelocity_58 0.007765521388992409 -0.007127282899392127 0.0008704679913017926
  translation -4.007423197087718 -0.2985708641668992 0.2894843657740887
  rotation -0.46379298847363676 -0.032034444095163775 -0.8853642517259249 0.06606964011645107
  name "GOALKEEPER_1"
  controller "SupervisorGoalkeeperController"
  controllerArgs [
    "1"
    "0"
  ]
  supervisor TRUE
  synchronization FALSE
  jersey NaoJersey {
    jerseyTexture [
      "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/softbank/nao/protos/textures/nao_2_red.png"
    ]
  }
}
TexturedBackground {
  texture "stadium"
}
TexturedBackgroundLight {
  castShadows FALSE
}
RobotstadiumSoccerField {
  frame1Color 0 0 1
  frame2Color 1 0 0
}
